monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/delegates/TemplatesDelegate.java

```diff
@@ -20,29 +20,147 @@
 
 package org.wso2.ei.dashboard.micro.integrator.delegates;
 
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+
 import org.wso2.ei.dashboard.core.commons.Constants;
-import org.wso2.ei.dashboard.core.db.manager.DatabaseManager;
-import org.wso2.ei.dashboard.core.db.manager.DatabaseManagerFactory;
+import org.wso2.ei.dashboard.core.commons.utils.HttpUtils;
+import org.wso2.ei.dashboard.core.commons.utils.ManagementApiUtils;
+import org.wso2.ei.dashboard.core.data.manager.DataManager;
+import org.wso2.ei.dashboard.core.data.manager.DataManagerSingleton;
+import org.wso2.ei.dashboard.core.exception.ManagementApiException;
 import org.wso2.ei.dashboard.core.rest.delegates.ArtifactDelegate;
 import org.wso2.ei.dashboard.core.rest.model.Ack;
+import org.wso2.ei.dashboard.core.rest.model.ArtifactDetails;
 import org.wso2.ei.dashboard.core.rest.model.ArtifactUpdateRequest;
 import org.wso2.ei.dashboard.core.rest.model.Artifacts;
+import org.wso2.ei.dashboard.core.rest.model.ArtifactsInner;
+import org.wso2.ei.dashboard.core.rest.model.ArtifactsResourceResponse;
+import org.wso2.ei.dashboard.micro.integrator.commons.DelegatesUtil;
+import org.wso2.ei.dashboard.micro.integrator.commons.Utils;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
-
+import java.util.concurrent.atomic.AtomicBoolean;
 /**
  * Delegate class to handle requests from templates page.
  */
 public class TemplatesDelegate implements ArtifactDelegate {
     private static final Logger logger = LogManager.getLogger(TemplatesDelegate.class);
-    private final DatabaseManager databaseManager = DatabaseManagerFactory.getDbManager();
+    private static final DataManager DATA_MANAGER = DataManagerSingleton.getDataManager();
+    private static List<ArtifactsInner> searchedList;
+    private static String prevSearchKey = null;
+    private static int count;
 
     @Override
-    public Artifacts getArtifactsList(String groupId, List<String> nodeList) {
-        logger.debug("Fetching templates from database.");
-        return databaseManager.fetchArtifacts(Constants.TEMPLATES, groupId, nodeList);
+    public ArtifactsResourceResponse getPaginatedArtifactsResponse(String groupId, List<String> nodeList, 
+        String searchKey, String lowerLimit, String upperLimit, String order, String orderBy, String isUpdate) 
+        throws ManagementApiException {
+        
+        logger.debug("Fetching Searched Templates from MI.");
+        int fromIndex = Integer.parseInt(lowerLimit);
+        int toIndex = Integer.parseInt(upperLimit);
+        boolean isUpdatedContent = Boolean.parseBoolean(isUpdate);
+
+        logger.debug("prevSearchkey :" + prevSearchKey + ", currentSearchkey:" + searchKey);
```

Comment:
Since we are doing a string concatenation let's have a `logger.isDebugEnabled()` check.